modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/execution/MethodArgumentResolverDelegate.java

```diff
@@ -192,8 +192,9 @@ private void initArgumentResolvers() {
         argumentResolver = ERROR_ARGUMENT_RESOLVER;
       } else if (SourceCallbackContext.class.equals(parameterType)) {
         argumentResolver = SOURCE_CALLBACK_CONTEXT_ARGUMENT_RESOLVER;
-      } else if (isParameterContainer(annotations.keySet(), typeLoader.load(parameterType)) &&
-          !((ParameterGroup) annotations.get(ParameterGroup.class)).showInDsl()) {
+      } else if (annotations.keySet().contains(ParameterGroup.class)
```

Comment:
I find it surprising that this moves the needle, specially since this only happens ones per MuleContext, but I choose to believe

Reply:
it avoids calling the `typeLoader.load` in some cases, which may be expensive.